src/compas_viewer/layout/propertyform.py

```diff
@@ -0,0 +1,425 @@
+from functools import partial
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Optional
+
+from compas.colors import Color
+from compas.geometry import Rotation
+from compas.geometry import Scale
+from compas.geometry import Transformation
+from compas.geometry import Translation
+from PySide6.QtWidgets import QCheckBox
+from PySide6.QtWidgets import QDoubleSpinBox
+from PySide6.QtWidgets import QHBoxLayout
+from PySide6.QtWidgets import QLabel
+from PySide6.QtWidgets import QLayout
+from PySide6.QtWidgets import QSpinBox
+from PySide6.QtWidgets import QVBoxLayout
+from PySide6.QtWidgets import QWidget
+
+from compas_viewer.scene.geometryobject import GeometryObject
+from compas_viewer.scene.meshobject import MeshObject
+from compas_viewer.scene.sceneobject import ViewerSceneObject
+
+from .collapsiblebox import CollapsibleBox
+
+if TYPE_CHECKING:
+    from compas_viewer.viewer import Viewer
+
+
+class Propertyform(QWidget):
+    """
+    Class for displaying basic properties of a scene object in a form.
+    Propertyform is an abstract layout class that could be placed the sidedock.
+
+    Parameters
+    ----------
+    obj : :class:`compas_viewer.scene.ViewerSceneObject`, optional
+        The object to be displayed initially.
+    on_select : bool, optional
+        If ``True``, the form will be updated when the object is selected.
+        Defaults to ``True``.
```

Comment:
would remove this. we tend to not mention defaults because you can see the value in the signature...